src/Featurizers/ShortGrainDropperFeaturizer.h

```diff
@@ -0,0 +1,255 @@
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType                          = StandardTransformer<std::vector<std::string>, bool>;
+    using GrainsSet                         = std::unordered_set<std::string>;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    explicit ShortGrainDropperTransformer(GrainsSet grainsToDrop);
+    explicit ShortGrainDropperTransformer(Archive &ar);
+
+    ~ShortGrainDropperTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(ShortGrainDropperTransformer);
+
+    bool operator==(ShortGrainDropperTransformer const &other) const;
+
+    void save(Archive &ar) const override;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    GrainsSet const                         _grainsToDrop;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+     // MSVC has problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        callback(_grainsToDrop.find(Traits<decltype(input)>::ToString(input)) != _grainsToDrop.end());
+    }
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         ShortGrainDropperEstimator
+///  \brief         Estimator to determine which grain to drop
+///
+template <
+    size_t MaxNumTrainingItemsV = std::numeric_limits<size_t>::max()
+>
+class ShortGrainDropperEstimator : public TransformerEstimator<std::vector<std::string>, bool> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType                          = TransformerEstimator<std::vector<std::string>, bool>;
+    using TransformerType                   = ShortGrainDropperTransformer;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    ShortGrainDropperEstimator(
+        AnnotationMapsPtr pAllColumnAnnotations,
+        size_t colIndex,
+        std::uint32_t windowSize,
+        std::vector<std::uint32_t> lags,
+        std::uint32_t maxHorizon,
+        nonstd::optional<std::uint32_t> cv
```

Comment:
I don't know what these variable names mean. Consider making the names more descriptive or add comments so that the caller has an idea of what to expect.